### ISSUE TITLE
False positive in block_in_if_condition_stmt: block generated by macro

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#3482]
```rust
fn main() {
    let option = Some(true);

    if option.unwrap_or_else(|| unimplemented!()) {
        unimplemented!()
    } else {
        unimplemented!()
    }
}
```

```console
warning: in an 'if' condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a 'let'
 --> src/main.rs:4:33
  |
4 |     if option.unwrap_or_else(|| unimplemented!()) {
  |                                 ^^^^^^^^^^^^^^^^
  |
  = note: #[warn(clippy::block_in_if_condition_stmt)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#block_in_if_condition_stmt
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

It doesn't look like there is a complex block or closure with block in the if-condition, although maybe the `unimplemented!()` macro expands to a block. The same lint does not trigger on `|| unimplemented()` in the same position, so I believe it should also not trigger for `|| unimplemented!()`.

I believe this is a different false positive from #1141.

---

*`clippy 0.0.212 (b2601be 2018-11-27)`*

**Comment by oli-obk** [comment]
just needs a macro check
